### PR TITLE
[ServiceBus] AAD samples and samples/README update

### PR DESCRIPTION
**Pull request by yunhaoling**

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/samples/README.md [hunk 1]
```diff
@@ -11,11 +11,26 @@ urlFragment: servicebus-samples
...
+    - Receive messages
+- [receive_peek.py](./sync_samples/receive_peek.py) ([async_version](./async_samples/receive_peek_async.py)) - Examples to peek messages from a service bus queue:
+    - Peek messages
+- [receive_deferred_message_queue.py](./sync_samples/receive_deferred_message_queue.py) ([async_version](./async_samples/receive_deferred_message_queue_async.py)) - Examples to defer received messages and receive deferred messages from a service bus queue:
+    - Defer received messages
+    - Receive deferred messages
+- [receive_iterator_queue.py](./sync_samples/receive_iterator_queue.py) ([async_version](./async_samples/receive_iterator_queue_async.py)) - Examples to receive messages from a service bus queue by iterating over ServiceBusReceiver:
+    - Receive messages by iterating over ServiceBusReceiver
+- [session_send_receive.py](./sync_samples/session_send_receive.py) ([async_version](./async_samples/session_send_receive_async.py)) - Examples to send messages to and received messages from a session-enabled service bus queue:
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/samples/README.md:28]:
```suggestion
- [session_send_receive.py](./sync_samples/session_send_receive.py) ([async_version](./async_samples/session_send_receive_async.py)) - Examples to send messages to and receive messages from a session-enabled service bus queue:
```

File: sdk/servicebus/azure-servicebus/samples/README.md [hunk 2]
```diff
@@ -11,11 +11,26 @@ urlFragment: servicebus-samples
...
+    - Send messages to a session-enabled queue
+    - Received messages from session-enabled queue
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/samples/README.md:30]:
```suggestion
    - Receive messages from session-enabled queue
```

File: sdk/servicebus/azure-servicebus/samples/sync_samples/client_identity_authentication.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from azure.servicebus import ServiceBusClient, Message
+from azure.identity import EnvironmentCredential
+
+
+FULLY_QUALIFIED_NAMESPACE = os.environ['SERVICE_BUS_NAMESPACE']
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+credential = EnvironmentCredential()
+
+# Note: One has other options to specify the credential.  For instance, DefaultAzureCredential.
```
Comment (KieranBrantnerMagee) [inline sdk/servicebus/azure-servicebus/samples/sync_samples/client_identity_authentication.py:48]: Did you want to put this disclaimer in async sample too?

Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/samples/sync_samples/client_identity_authentication.py:48]: added in async sample

**Pull request merged**